services/blockchain-indexer/shared/dataService/business/delegates.js

```diff
@@ -73,14 +71,26 @@ const getAllDelegates = async () => {
 	return delegates;
 };
 
-const getDelegates = async (addresses) => {
+const getDelegates = async (params) => {
+	const addresses = params.address ? [params.address] : params.addresses;
+
 	const delegates = await BluebirdPromise.map(
 		addresses,
 		async address => {
 			const delegate = await requestConnector(
 				'dpos_getDelegate',
```

Comment:
Please avoid invoking the direct endpoint. Instead create and use the interface, if not exists.
```suggestion
				'getDelegate',
```